packages/core/src/routes/interaction/verifications/mandatory-user-profile-validation.ts

```diff
@@ -175,19 +190,22 @@ export default async function validateMandatoryUserProfile(
       : // eslint-disable-next-line unicorn/consistent-destructuring -- have to infer the accountId existence by event !== register
         await userQueries.findUserById(interaction.accountId);
 
-  const missingProfileSet = getMissingProfileBySignUpIdentifiers({ signUp, user, profile });
+  const requiredProfileSet = getMissingProfileBySignUpIdentifiers({ signUp, user, profile });
```

Comment:
`requiredProfileSet` or `missingProfileSet`? I'm a bit confused

Reply:
the idea is keep a similar naming

Reply:
As discussed offline, we can improve this by moving the `assertThat` into the `getMissingProfileBySignUpIdentifiers` function so that the return value and the logic in the current function can be simplified.
We will going to revise that in a follow-up PR, though. Resoving this one